jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-client/src/main/java/org/jbpm/workbench/cm/client/roles/CaseRolesPresenter.java

```diff
@@ -48,42 +56,150 @@ public String getTittle() {
 
     @Override
     protected void clearCaseInstance() {
-        view.disableNewRoleAssignments();
         view.removeAllRoles();
     }
 
     @Override
     protected void loadCaseInstance(final CaseInstanceSummary cis) {
-        view.addUser(cis.getOwner(), translationService.format(OWNER));
-
-        setupRoleAssignments(cis);
-
-        setupNewRoleAssignments(cis);
-
-        view.setupPagination();
+        if (passFilter(Arrays.asList(cis.getOwner()), null)) {
+            view.addOwnerAssignment(translationService.format(OWNER), cis.getOwner());
+        }
```

Comment:
@nmirasch Yes, I understand now. This comment is actually related to that other comment I made about sending requests to kie server every time filtering is changed.
When I made this comment I didn't expect that loadCaseInstance() method will be used repeatedly, but rather only once i.e. when the case is loaded. In that case, this evaluation wouldn't make sense.
It would be good if it would be possible to somehow decouple loading of case roles from handling their lists in various filter tabs, as I think all the data you need can be saved are re-used without calling this method repeatedly.